oracle/types/provider.go

```diff
@@ -0,0 +1,103 @@
+package types
+
+import (
+	"fmt"
+	"sync"
+)
+
+type (
+	// ProviderTicker is the interface for the ticker that provider's utilize/return.
+	ProviderTicker interface {
+		fmt.Stringer
+
+		// GetProvider returns the provider for the ticker.
+		GetProvider() string
```

Comment:
Why does the ProviderTicker need to know which provider it's associated with? I can't see anywhere that the `GetProvider` method is called, and when `ProviderTicker` is used it's generally in the context of a `ProviderTickers` which is just an association between a single Provider and its offchain Tickers.